geonode_mapstore_client/client/js/epics/index.js

```diff
@@ -33,36 +32,25 @@ import get from 'lodash/get';
  */
 import { showCoordinateEditorSelector } from '@mapstore/framework/selectors/controls';
 
-/**
- * When a user selects a layer, the app checks for layer editing permission.
- */
-export const _setFeatureEditPermission = (action$, { getState } = {}) =>
-    action$.ofType(SELECT_NODE).filter(({ nodeType }) => nodeType === "layer" && !getConfigProp("disableCheckEditPermissions"))
-        .switchMap(() => {
-            const layer = getSelectedLayer(getState() || {});
-            return layer ? layerEditPermissions(layer)
-                .map(permissions => setPermission(permissions))
-                .startWith(setPermission({ canEdit: false }))
-                .catch(() => Rx.Observable.empty()) : Rx.Observable.of(setPermission({ canEdit: false }));
-        });
-/**
- * When a user selects a layer, the app checks for style editing permission.
- * INIT_STYLE_SERVICE si needed for map editing, it ensures an user has permission to edit style of a specific layer retrieved from catalog
- */
-export const _setStyleEditorPermission = (action$, { getState } = {}) =>
-    action$.ofType(INIT_STYLE_SERVICE, SELECT_NODE)
-        .filter(({ nodeType }) =>
-            nodeType && nodeType === "layer" && !getConfigProp("disableCheckEditPermissions")
-            || !nodeType && !getConfigProp("disableCheckEditPermissions"))
-        .switchMap((action) => {
-            const layer = getSelectedLayer(getState() || {});
-            return layer
-                ? styleEditPermissions(layer)
-                    .map(({ canEdit }) => setEditPermissionStyleEditor(canEdit))
-                    .startWith(setEditPermissionStyleEditor(action.canEdit))
-                    .catch(() => Rx.Observable.empty())
-                : Rx.Observable.of(setEditPermissionStyleEditor(false));
-        });
+
+export const checkFeatureAndEditorPermissions = (action$, { getState } = {}) =>
+action$.ofType(SELECT_NODE, INIT_STYLE_SERVICE)
+    .filter(({ nodeType }) => nodeType && nodeType === "layer" && !getConfigProp("disableCheckEditPermissions")
+        || !nodeType && !getConfigProp("disableCheckEditPermissions"))
+    .switchMap(() => {
+        const state = getState() || {}
+        const layer = getSelectedLayer(state);
+        return layer ? Rx.Observable.defer(() => getLayerByName(layer?.name))
+        .map(data => {
+            const permissions = data?.layers[0]?.perms || [];
+            const canEditStyles = permissions.includes("change_layer_style")
+            const canEdit = permissions.includes("change_layer_data");
+            return {canEdit, canEditStyles, permissions}
+        })
+        .map(({canEdit, canEditStyles, permissions}) => (setPermission({canEdit}), setEditPermissionStyleEditor(canEditStyles),setUserResourcePermissions(permissions)))
```

Comment:
It seems that with this structure setPermission and setEditPermissionStyleEditor are not tirggered but only setUserResourcePermissions.

Proposed solution:

```js
.switchMap(layer => {
    const permissions = layer?.perms || [];
    const canEditStyles = permissions.includes("change_layer_style")
    const canEdit = permissions.includes("change_layer_data");
    return Observable.of(
        setPermission({ canEdit }),
        setEditPermissionStyleEditor(canEditStyles),
        setUserResourcePermissions(permissions)
    );
})
```